service/Checkout.php

```diff
@@ -24,10 +24,56 @@
 
 namespace Adyen\PrestaShop\service;
 
+use Adyen\PrestaShop\service\adapter\classes\ServiceLocator;
+use Cart;
+
 class Checkout extends \Adyen\Service\Checkout
 {
+    const PAYMENT_METHOD_STEP = 'checkout-payment-step';
+    const DELIVERY_STEP = 'checkout-delivery-step';
+    const IS_REACHABLE = 'step_is_reachable';
+    const IS_COMPLETE = 'step_is_complete';
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+
     public function __construct(Client $client)
     {
         parent::__construct($client);
+        $this->logger = ServiceLocator::get('Adyen\PrestaShop\service\Logger');
+    }
+
+    /**
+     * Check if the payment method step is next in the checkout process
+     *
+     * @param Cart $cart
+     *
+     * @return bool
+     */
+    public function isPaymentMethodStepNext(Cart $cart)
+    {
+        try {
+            $checkoutSessionData = \Db::getInstance()->getValue(
+                'SELECT checkout_session_data FROM ' . _DB_PREFIX_ . 'cart WHERE id_cart = ' . (int)$cart->id
+            );
+
+            $jsonData = json_decode($checkoutSessionData, true);
+
+            // If delivery step is complete and payment method is reachable
+            if ($jsonData[self::DELIVERY_STEP][self::IS_COMPLETE] &&
```

Comment:
here as well. Always make sure that these keys `self::DELIVERY_STEP` `self::IS_COMPLETE` `self::PAYMENT_METHOD_STEP` `self::IS_REACHABLE` do exists otherwise it can create some nasty errors for you that will throw a warning and won't be caught by the try catch

Reply:
https://github.com/Adyen/adyen-prestashop/pull/207/commits/7a886af752f3521ac358b05509f8f9d2f2a9c41d